src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,101 @@
+package core.basesyntax.service;
+
+import core.basesyntax.model.User;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+class RegistrationServiceImplTest {
+
+    private RegistrationServiceImpl registrationService;
+
+    @BeforeEach
+    public void setUp() {
+        registrationService = new RegistrationServiceImpl();
+    }
+
+    @Test
+    void register_nullUser_notOk() {
+        registrationService.register(null);
+    }
+
+    @Test
+    void register_shortLogin_notOk() {
+        User user = new User();
+        user.setLogin("abc");
+        user.setPassword("password");
+        user.setAge(20);
+        registrationService.register(user);
+    }
+
+    @Test
+    void register_duplicateLogin_notOk() {
+        User user1 = new User();
+        user1.setLogin("uniqueLogin");
+        user1.setPassword("password");
+        user1.setAge(20);
+        registrationService.register(user1);
+
+        User user2 = new User();
+        user2.setLogin("uniqueLogin");
+        user2.setPassword("password123");
+        user2.setAge(21);
+        registrationService.register(user2);
+    }
+
+    @Test
+    void register_shortPassword_notOk() {
+        User user = new User();
+        user.setLogin("validLogin");
+        user.setPassword("123");
+        user.setAge(20);
+        registrationService.register(user);
+    }
+
+    @Test
+    void register_simplePassword_notOk() {
+        User user = new User();
+        user.setLogin("validLogin");
+        user.setPassword("simple");
+        user.setAge(20);
+        RegistrationException registrationException = Assertions.assertThrows(RegistrationException.class, () -> registrationService.register(user));
+        Assertions.assertEquals("The password must be complex!", registrationException.getMessage());
```

Comment:
It's good that you are checking the exception message here. However, the requirement to have a complex password is not mentioned in the task description. Therefore, this test case might not be necessary.